### ISSUE TITLE
Update MPICH 4.0.3 to 4.1 disabled by DBCSR library build

### ISSUE THREAD

**Issue by mkrack** [cp2k/cp2k#2808]
An update from MPICH 4.0.3 to 4.1.x is currently disabled by the build of the DBCSR library which is required for the CP2K build. The problem is already described by the DBCSR issue [#661](https://github.com/cp2k/dbcsr/issues/661) and by the comment in the routine header of `mp_alloc_mem_`:
```
      SUBROUTINE mp_alloc_mem_${nametype1}$ (DATA, len, stat)
      !! Allocates an array, using MPI_ALLOC_MEM ... this is hackish
      !! as the Fortran version returns an integer, which we take to be a C_PTR

         ${type1}$, DIMENSION(:), POINTER, CONTIGUOUS :: DATA
         !! data array to allocate
         INTEGER, INTENT(IN)                      :: len
         !! length (in data elements) of data array allocation
         INTEGER, INTENT(OUT), OPTIONAL           :: stat
         !! allocation status result

         INTEGER                                  :: size, ierr, length, &
                                                     mp_info, mp_res
         INTEGER(KIND=MPI_ADDRESS_KIND)           :: mp_size
         TYPE(C_PTR)                              :: mp_baseptr

         length = MAX(len, 1)
         CALL MPI_TYPE_SIZE(${mpi_type1}$, size, ierr)
         mp_size = INT(length, KIND=MPI_ADDRESS_KIND)*size
         IF (mp_size .GT. mp_max_memory_size) THEN
            DBCSR_ABORT("MPI cannot allocate more than 2 GiByte")
         END IF
         mp_info = MPI_INFO_NULL
         CALL MPI_ALLOC_MEM(mp_size, mp_info, mp_baseptr, mp_res)
         CALL C_F_POINTER(mp_baseptr, DATA, (/length/))
         IF (PRESENT(stat)) stat = mp_res
      END SUBROUTINE mp_alloc_mem_${nametype1}$
```
The routine uses the Fortran 2008 types, i.e. `C_PTR` for `mp_baseptr` implying `USE mpi_f08`, while still the old Fortran MPI module is employed by `USE MPI` which returns an `INTEGER` of `MPI_ADDRESS_KIND` for `mp_baseptr`. It seems that from MPICH 4.1 transitional interfaces have been dropped eventually which allowed the compilation in the past.

**Comment by fstein93** [comment]
I may have a look at it because I have already some experiences in that respect with CP2K.

**Comment by mkrack** [comment]
Great, thanks @fstein93

**Comment by alazzaro** [comment]
Please try again, now DBCSR support MPI F08 interface.

**Comment by mkrack** [comment]
The issue is resolved with DBCSR release 2.6.0.
Thanks to @alazzaro and @fstein93
